File: python/hamming/hamming.py
def diff(pair):
    """Returns True when pair elements are , otherwise False."""
    x, y = pair
    return x != y


def distance(strand_a, strand_b):
    """Returns hamming distance of two dna strands."""
    if len(strand_a) != len(strand_b):
        raise ValueError("Strands must be of the same length")
    return len(filter(diff, zip(strand_a, strand_b)))

File: ruby/hamming/hamming.rb
# frozen_string_literal: true

# Calculate hamming distance between DNA sequences
class Hamming
  def self.compute(seq1, seq2)
    if seq1.length != seq2.length
      raise ArgumentError, 'DNA sequences must be the same length'
    end

    seq1.chars.zip(seq2.chars).count { |c1, c2| c1 != c2 }
  end
end
